src/ServiceControl/Infrastructure/Mail/CustomChecksMailNotification.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace ServiceControl.Infrastructure.Mail
+{
+    using System.Net.Mail;
+    using System.Threading.Tasks;
+    using Alerting;
+    using Contracts.CustomChecks;
+    using DomainEvents;
+    using Raven.Client;
+    using ServiceBus.Management.Infrastructure.Settings;
+
+    class CustomChecksMailNotification : IDomainHandler<CustomCheckFailed>, IDomainHandler<CustomCheckSucceeded>
+    {
+        readonly IDocumentStore store;
+        readonly string emailDropFolder;
+
+        public CustomChecksMailNotification(IDocumentStore store, Settings settings)
+        {
+            this.store = store;
+            emailDropFolder = settings.EmailDropFolder;
+        }
+
+        public async Task Handle(CustomCheckFailed domainEvent)
+        {
+            var settings = await LoadSettings().ConfigureAwait(false);
+            if (settings == null || !settings.AlertingEnabled)
+            {
+                return;
+            }
+
+            var smtpClient = CreateSmtpClient(settings);
+
+            var mailMessage = new MailMessage(settings.From, settings.To, "Health check failed",
+                $"Health check {domainEvent.Category}: {domainEvent.CustomCheckId} failed at {domainEvent.FailedAt}. Failure reason {domainEvent.FailureReason}");
+
+            await smtpClient.SendMailAsync(mailMessage).ConfigureAwait(false);
+        }
+
+        SmtpClient CreateSmtpClient(AlertingSettings settings)
+        {
+            if (emailDropFolder != null)
+            {
+                return new SmtpClient
+                {
+                    PickupDirectoryLocation = emailDropFolder,
+                    DeliveryMethod = SmtpDeliveryMethod.SpecifiedPickupDirectory
+                };
+            }
+            return new SmtpClient(settings.SmtpServer, settings.SmtpPort ?? 25)
```

Comment:
WHy not use mailsettings in app.config? IMHO this should not be done in code:

```xml
<system.net>
  <mailSettings>
    <smtp from="mail@domain.com">
      <network host="myserver.com" port="25" defaultCredentials="true" />
    </smtp>
  </mailSettings>
</system.net>
```

This way you can support authentication methods.

Reply:
Good question. We assumed all the config is going to be in RavenDB and available via ServicePulse. @tmasternak what do you think about the configuration-based approach?

Reply:
I would keep this in the UI. If users want to test if notifications are working they can trigger that from the UI but would not be able to see the settings.